File: DSA With Java/Array/Add1InArray.java
import java.util.Scanner;

//arr=[5,7,3,8]
//ans =5739
class Add1InArray {
    public int getArraySum(int n, int[] arr) {
        int ans = 0, rev = 0, temp = 0, i = 0;
        for (i = 0; i < n; i++) {
           
            rev = rev * 10 + arr[i];
        }

        return rev + 1;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n, i;
        System.out.println("Enter size of array");
        n = scanner.nextInt();
        int[] arr = new int[n];

        System.out.println("Enter element of array");
        for (i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        Add1InArray objectAdd1InArray = new Add1InArray();
        System.out.println("After add 1" + objectAdd1InArray.getArraySum(n, arr));
    }
}

File: Array/Add1inArray.cpp
#include "bits/stdc++.h"
using namespace std;
// Leetcode: https://leetcode.com/problems/plus-one/
// class add1inArray{
//   public:
//   vector<int> add1(int *arr, int n){
//     int temp=0,i,rem=0;
//     for ( i = 0; i <n; i++){
//         temp=(temp*10)+arr[i];
//     }
//     temp=temp+1;
//      vector<int>v;
//     for ( i = 0; i < n; i++) {
//         rem=temp%10;
//         v.push_back(rem);
//         temp=temp/10;
//     }
//     reverse(v.begin(),v.end());
//     return v;
//   }
// };
class add1inArray
{
public:
    vector<int> add1InArray(int *arr, int n)
    {
        int temp = 0;
        for (int i = 0; i < n; i++)
        {
            temp = temp * 10 + arr[i];
        }
        vector<int> ans;
        temp = temp + 1;
        for (int i = 0; i < n; i++)
        {
            ans.push_back(temp % 10);
            temp = temp / 10;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};
int main(int argc, char const *argv[])
{
    add1inArray obj;
    int i, n;
    cout << "Enter size of array: " << endl;
    cin >> n;
    int arr[n];
    cout << "Enter Elemement of Array : " << endl;
    for (i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    cout << "After add 1 in array : " << endl;
    vector<int> ans = obj.add1InArray(arr, n);

    for (auto x : ans)
    {
        cout << x << " ";
    }

    return 0;
}

/*
Input : [4,6,7,7,8]
Output : 46779
*/